api_docs/message-formatting.md

```diff
@@ -0,0 +1,44 @@
+# Message formatting
+
+Zulip supports an extended version of Markdown for messages, as well as
+some HTML level special behavior. This article is currently a changelog
+for updates to these features.
+
+**Changes**: In Zulip 4.0 (feature level 24), the rarely used `!avatar()`
+and `!gravatar()` markup syntax, which was never documented and had an
+inconsistent syntax, were removed.
+
+## Code blocks
+
+**Changes**: As of Zulip 4.0 (feature level 33), [code blocks][help-code]
+can have a `data-code-language` attribute attached to the outer HTML
+`div` element, which records the programming language that was selected
+for syntax highlighting. This field is used in the
+[playgrounds][help-playgrounds] feature for code blocks.
+
+## Global times
+
+**Changes**: In Zulip 3.0 (feature level 8), added [global time
+mentions][help-global-time] to supported Markdown message formatting
+features.
+
+## Mentions
+
+**Changes**: In Zulip 9.0 (feature level 246), `channel` was added
+to the supported [wildcard][help-mention-all] options used in the
+[mentions][help-mentions] Markdown message formatting feature. This
+deprecates the `stream` wildcard option, which is still supported
+for backwards compatibility.
+
+## Spoilers
+
+**Changes**: In Zulip 3.0 (feature level 15), added
+[spoilers][help-spoilers] to supported Markdown message formatting
+features.
+
+[help-code]: /help/code-blocks
```

Comment:
Let's add a "Related articles" section; initially can link to the send/render API pages, and the main Markdown page in /help/.

Reply:
Done in updated changes.